### PR TITLE
Add explanation mounting named volumes for `podman run`

### LINKED ISSUE #3811: [doc] `podman run` doesn't elaborate about named volumes
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

Neither `--volume` nor `--mount` in [here](https://github.com/containers/libpod/blob/master/docs/podman-run.1.md) talk about named volumes. They only elaborate mounting a external data source with absolute path (which might be a cause of podman-compose attampts to create named volumes and bind mount them, and [might lead to data loss](https://github.com/muayyad-alsadi/podman-compose/issues/13#issuecomment-521123163)).

In Docker, both `-v` and `-m` can mount named volumes. Just give it's name. Podman handles this the same way as well.

**Describe the results you received:**
No documentation [here](https://github.com/containers/libpod/blob/master/docs/podman-run.1.md) talks about how to mount named volumes

**Describe the results you expected:**
Elaborate how to use `-v` and `-m` in `podman run` to mount named volumes in a managed state.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
(paste your output here)
```

**Output of `podman info --debug`:**

```
(paste your output here)
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

### PR DESCRIPTION
**Pull request by xcffl**
Fixes #3811

### CODE REVIEW

**Comment by xcffl** [comment]
Also adds explanation about configuring fuse-overlayfs on rootless mode

File: docs/podman-run.1.md [hunk 1]
```diff
@@ -809,18 +811,21 @@ Set the UTS mode for the container
...
 **NOTE**: the host mode gives the container access to changing the host's hostname and is therefore considered insecure.
 
-**--volume**, **-v**[=*[HOST-DIR:CONTAINER-DIR[:OPTIONS]]*]
+**--volume**, **-v**[=*[HOST-DIR-OR-VOUME-NAME:CONTAINER-DIR[:OPTIONS]]*]
 
 Create a bind mount. If you specify, ` -v /HOST-DIR:/CONTAINER-DIR`, podman
 bind mounts `/HOST-DIR` in the host to `/CONTAINER-DIR` in the podman
-container. The `OPTIONS` are a comma delimited list and can be:
+container. Similarly, `-v VOLUME-NAME:/CONTAINER-DIR` will mount the volume 
+to the container. If no such named volume, podman will create one.
```
Comment (mheon) [inline docs/podman-run.1.md]: Add "exists" after "if no such named volume"

File: docs/tutorials/rootless_tutorial.md [hunk 2]
```diff
@@ -22,6 +22,8 @@ The [slirp4netns](https://github.com/rootless-containers/slirp4netns) package pr
 
 When using Podman in a rootless environment, it is recommended to use fuse-overlayfs rather than the VFS file system.  Installing the fuse3-devel package gives Podman the dependencies it needs to install, build and use fuse-overlayfs in a rootless environment for you.  The fuse-overlayfs project is also available from [GitHub](https://github.com/containers/fuse-overlayfs).  This especially needs to be checked on Ubuntu distributions as fuse-overlayfs is not generally installed by default.
 
+After installing fuse-overlayfs, in `$HOME/.config/containers/storage.conf` change `driver` under `[storage]` to `"overlay"`.
```
Comment (mheon) [inline docs/tutorials/rootless_tutorial.md]: This should be done automatically if you've never used Podman before, so this should probably say it "may be necessary to change the `driver` option..." or similar

**Comment by mheon** [comment]
Both commits need signoffs (can add via `git commit --amend -s`, but you'll have to do a `rebase -i` to hit both)

**Comment by rhatdan** [comment]
LGTM
Thanks @xcffl

**Comment by rhatdan** [comment]
Looks like the tests are complaining about some additional whitespace?

File: docs/podman-run.1.md [hunk 3]
```diff
@@ -809,18 +811,21 @@ Set the UTS mode for the container
 
 **NOTE**: the host mode gives the container access to changing the host's hostname and is therefore considered insecure.
 
-**--volume**, **-v**[=*[HOST-DIR:CONTAINER-DIR[:OPTIONS]]*]
+**--volume**, **-v**[=*[HOST-DIR-OR-VOUME-NAME:CONTAINER-DIR[:OPTIONS]]*]
 
 Create a bind mount. If you specify, ` -v /HOST-DIR:/CONTAINER-DIR`, podman
 bind mounts `/HOST-DIR` in the host to `/CONTAINER-DIR` in the podman
-container. The `OPTIONS` are a comma delimited list and can be:
+container. Similarly, `-v VOLUME-NAME:/CONTAINER-DIR` will mount the volume
```
Comment (TomSweeneyRedHat) [inline docs/podman-run.1.md]: maybe "mount the volume" to "mount the volume in the host" to be absolutely clear?

File: docs/podman-run.1.md [hunk 4]
```diff
@@ -1087,8 +1092,8 @@ $ podman run -v /var/db:/data1 -i -t fedora bash
 ```
 
 Using --mount flags, To mount a host directory as a container folder, specify
-the absolute path to the directory and the absolute path for the container
-directory:
+the absolute path or volume name to the directory and the absolute path for
```
Comment (TomSweeneyRedHat) [inline docs/podman-run.1.md]: I think this should be "the absolute path to the directory or the volume name,".

File: docs/podman-run.1.md [hunk 5]
```diff
@@ -809,18 +811,21 @@ Set the UTS mode for the container
...
 **NOTE**: the host mode gives the container access to changing the host's hostname and is therefore considered insecure.
 
-**--volume**, **-v**[=*[HOST-DIR:CONTAINER-DIR[:OPTIONS]]*]
+**--volume**, **-v**[=*[HOST-DIR-OR-VOUME-NAME:CONTAINER-DIR[:OPTIONS]]*]
 
 Create a bind mount. If you specify, ` -v /HOST-DIR:/CONTAINER-DIR`, podman
 bind mounts `/HOST-DIR` in the host to `/CONTAINER-DIR` in the podman
-container. The `OPTIONS` are a comma delimited list and can be:
+container. Similarly, `-v VOLUME-NAME:/CONTAINER-DIR` will mount the volume 
+to the container. If no such named volume exists, podman will create one.
```
Comment (TomSweeneyRedHat) [inline docs/podman-run.1.md]: a couple of "podman" instances that I'd prefer as "Podman".  One was pre-existing.

File: docs/tutorials/rootless_tutorial.md [hunk 6]
```diff
@@ -22,6 +22,8 @@ The [slirp4netns](https://github.com/rootless-containers/slirp4netns) package pr
 
 When using Podman in a rootless environment, it is recommended to use fuse-overlayfs rather than the VFS file system.  Installing the fuse3-devel package gives Podman the dependencies it needs to install, build and use fuse-overlayfs in a rootless environment for you.  The fuse-overlayfs project is also available from [GitHub](https://github.com/containers/fuse-overlayfs).  This especially needs to be checked on Ubuntu distributions as fuse-overlayfs is not generally installed by default.
 
+If podman is installed before fuse-overlayfs, it may be necessary to change `driver` option under `[storage]` to `"overlay"`.
```
Comment (TomSweeneyRedHat) [inline docs/tutorials/rootless_tutorial.md]: s/change `driver`/change the `driver`/

**Comment by TomSweeneyRedHat** [comment]
Few comments, thanks for the PR @xcffl !

**Comment by baude** [comment]
please add an appropriate commit message

**Comment by baude** [comment]
you need to run make validate on your commits and repush

File: docs/podman-run.1.md (hunk 5 above)
Comment (xcffl) [reply docs/podman-run.1.md]: > a couple of "podman" instances that I'd prefer as "Podman". One was pre-existing.

You mean it's better to substitute "podman" in the text to "Podman"? Other places in the documentation use "podman". If you think that's better, maybe we can replace all of them.

File: docs/podman-run.1.md [hunk 7]
```diff
@@ -809,18 +811,21 @@ Set the UTS mode for the container
 
 **NOTE**: the host mode gives the container access to changing the host's hostname and is therefore considered insecure.
 
-**--volume**, **-v**[=*[HOST-DIR:CONTAINER-DIR[:OPTIONS]]*]
+**--volume**, **-v**[=*[HOST-DIR-OR-VOUME-NAME:CONTAINER-DIR[:OPTIONS]]*]
```
Comment (TomSweeneyRedHat) [inline docs/podman-run.1.md]: Please add an example showing how the volume-name is used down below in the examples

Comment (xcffl) [reply docs/podman-run.1.md]: > Please add an example showing how the volume-name is used down below in the examples

I've already add an example a paragraph after. Maybe it should be more clear?

File: docs/podman-run.1.md (hunk 5 above)
Comment (TomSweeneyRedHat) [reply docs/podman-run.1.md]: If you've the time, then please do change all 'podman" instances to 'Podman' unless it's in a command example.

File: docs/podman-run.1.md (hunk 7 above)
Comment (TomSweeneyRedHat) [reply docs/podman-run.1.md]: It was just that you added this "or volume name" thought to the docs, but I didn't see any additional examples.  Maybe in the example below add a little verbiage and a second example?

To add a volume using a host directory:

{example}

To add a volume using the volume name:

{example}

**Comment by xcffl** [comment]
I think everything is fixed now

**Comment by mheon** [comment]
We're having some CI trouble so the Prow tests may fail to re-run for the merge. We'll see.

**Pull request merged**